packages/holocron/src/loadModule.node.js

```diff
@@ -37,66 +54,221 @@ function fetchModuleContent(moduleUrl) {
+    } else {
+      throw new Error(messages.join('\n'));
+    }
+
+    fallbackExternals.forEach((external) => {
+      addRequiredExternal(external);
+    });
   }
+}
 
-  const error = new Error(response.statusText);
-  error.response = response;
-  throw error;
+/**
+ * Fetches Module Config. Returns null if does not exist or an error has occured
+ * @param {string} baseUrl path to the assets
+ * @returns {Promise<object | null>}
+ */
+const fetchModuleConfig = async (baseUrl) => {
+  const moduleConfigUrl = `${baseUrl}module-config.json`
+
+  try {
+    const moduleConfigStr = await fetchAsset(moduleConfigUrl);
+    const moduleConfig = JSON.parse(moduleConfigStr);
+
+    return moduleConfig;
+  } catch (err) {
+    console.warn('Module Config failed to fetch and parse, external fallbacks will be ignored.', err)
+  }
+
+  return null;
 }
 
-export default async function loadModule(
+const loadModule = async (
   moduleName,
-  { node: { integrity, url } },
+  { node: { integrity, url }, baseUrl },
   onModuleLoad = () => null
-) {
+) => {
   try {
     assert(typeof moduleName === 'string', 'moduleName must be a string');
 
     if (isModuleInBlockList(url)) {
       throw new Error(`module at ${url} previously failed to load, will not attempt to reload.`);
     }
 
-    const moduleResponse = await fetchModuleContent(url);
-    let nodeModule;
+    const rootModule = global.getTenantRootModule?.();
 
-    try {
-      checkStatus(moduleResponse);
-      const moduleString = await moduleResponse.text();
-      if (process.env.NODE_ENV === 'production') {
-        const actualSRI = ssri.fromData(
-          moduleString,
-          { algorithms: ['sha256', 'sha384'] }
-        ).toString();
-
-        assert(integrity === actualSRI, `SRI for module at ${url} must match SRI in module map.\n Expected ${integrity}, received ${actualSRI}`);
-      }
-      nodeModule = requireFromString(moduleString, url);
-    } catch (err) {
-      if (process.env.NODE_ENV === 'development') {
-        console.warn('Holocron module "%s" at "%s" failed to execute.\n\t[Error Message]: "%s"\nPlease fix any errors and wait for it to be reloaded.', moduleName, url, err.message);
-      } else if (err.shouldBlockModuleReload !== false) {
-        addToModuleBlockList(url);
-        console.warn(`Holocron module at ${url} added to blocklist: ${err.message}`);
+    if (rootModule) {
+      const rootModuleConfig = rootModule.appConfig;
+      const { requiredExternals } = (await fetchModuleConfig(baseUrl)) || {};
+
+      if (requiredExternals) {
+        const {
+          providedExternals: rootProvidedExternals,
+          enableUnlistedExternalFallbacks,
+        } = rootModuleConfig;
+
+        validateRequiredExternals({
+          moduleName,
+          requiredExternals,
+          providedExternals: rootProvidedExternals || {},
+          enableUnlistedExternalFallbacks,
+        });
+        registerModuleUsingExternals(moduleName);
```

Comment:
this should come after onModuleLoad. if onModuleLoad fails this should not be sitting in the registry